Settings menu

twjvdhorst opened the pull request:
Done with the settings menu. Only thing missing is the fow button, but I can't get that to work

Commit:
scrollbar

nessemis committed:
rebased

nessemis committed:
merged settingsmenu

nessemis committed:
merged

nessemis committed:
fixed compile errors

nessemis committed:
fixed splash clicking

nessemis committed:
settings overlay

Commit:
added slider

Commit:
undid gamebreak

Commit:
removed usings

Commit:
Volume slider

Commit:
fixed slider

Commit:
slider

Commit:
fixed

Commit:
Changed slider dimensions

Commit:
Merge remote-tracking branch 'origin/Settings-Menu' into Settings-Menu

# Conflicts:
#	MeesGame/Gameplay/GameStates/SettingsMenuState.cs

Commit:
Functional Volume Slider

Commit:
Final changes

Deleted the fow button, didn't work. Added volume slider

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Settings-Menu

# Conflicts:
#	MeesGame/Gameplay/GameObjects/TileField.cs

jorisBarkema commented:
Would be better if instead of going back by pressing p, as it is now, you can press p to enter the settings menu from anywhere, and close it to return to where you called it from, but I think that last bit could be hard, not sure exactly how the UI works. At least press p to enter settings menu instead of just going back to menu should be doable, and I think that would be an improvement.

Custom resolution is very fancy though and volume slider is great!

jorisBarkema commented on MeesGame/MeesGame.cs:
```diff
@@ -28,13 +28,14 @@ protected override void LoadContent()
             base.LoadContent();
             screen = new Point(1440, 825);
-            windowSize = new Point(1024, 586);
+            //windowSize = new Point(1024, 586);
```
this comment seems redundant

jorisBarkema commented on MeesGame/Gameplay/GameObjects/TileField.cs:
```diff
@@ -10,8 +10,19 @@ public class TileField : GameObjectGrid, IDiscreteField, ITileField
+        public Point Start
+        {
+            get
+            {
```
Don;t think you need to change anything in tilefied

LisanneW replied:
Even makes the game crash if you try to run it.

LisanneW replied:
I think something went wrong while merging the master branch because this property was relocated

jensj12 replied:
Move the property back to where it was.

jorisBarkema commented:
I do not have enough confidence in myself to review the rest of this.

Commit:
Changes according to comments

Still looking into making it an overlay instead of an entire gamestate

Commit:
Pressing P now goes to the settings

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Settings-Menu

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Settings-Menu

# Conflicts:
#	MeesGame/Gameplay/GameStates/TitleMenuState.cs
#	MeesGame/Strings1.Designer.cs
#	MeesGame/UI/DefaultUIValues.settings
#	MeesGame/app.config

Commit:
conflict

jensj12 commented:
Since the resolution setting causes a few bugs, I think it's better to leave it out and just fix fullscreen mode.

LisanneW commented:
@nessemis or @twjvdhorst can you fix the issues with the resolution settings or put that in another pull request so this can be merged?

Commit:
Got rid of the buttons for windowsize/resolution

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Settings-Menu

Commit:
Changed startmenu

Moved "Create Level" button to "Begin Game", moved "Settings" button to
startmenu, where Create Level was

jensj12 commented on MeesGame/Gameplay/GameManagement/GameEnvironment.cs:
```diff
@@ -119,9 +125,9 @@ protected override void LoadContent()
+        if (inputHelper.KeyPressed(Keys.P))
         {
-            Exit();
+            gameStateManager.SwitchTo("SettingsMenuState");
```
Duplicate code. This is already in the gamestatemanager.

jensj12 commented on MeesGame/Gameplay/GameManagement/GameStateManager.cs:
```diff
@@ -53,7 +53,7 @@ public string PreviousGameState
     public void HandleInput(InputHelper inputHelper)
     {
-        if (inputHelper.KeyPressed(Keys.P)) SwitchTo("TitleMenuState");
+        if (inputHelper.KeyPressed(Keys.P)) SwitchTo("SettingsMenuState");
```
Add escape here. Either make it go to the settings menu or main menu.

jensj12 commented on MeesGame/Gameplay/GameStates/LoadMenuState.cs:
```diff
@@ -64,10 +65,18 @@ public LoadMenuState()
+            {
+                //reset the gamestate to open blank;
+                GameEnvironment.GameStateManager.GetGameState("EditorState").Reset();
+                GameEnvironment.GameStateManager.SwitchTo("EditorState");
```
First switch to the new game state, then reset the current game state.

jensj12 commented on MeesGame/UI/Components/Slider.cs:
```diff
@@ -0,0 +1,192 @@
+                if (Value > max)
+                    Value = max;
+                if (Value < min)
+                    Value = min;
```
Use MathHelper.Clamp(Value,min,max)

jensj12 commented on MeesGame/UI/DefaultUIValues.settings:
```diff
@@ -41,11 +41,18 @@
     </Setting>
     <Setting Name="RadioButtonDimen" Type="System.Int32" Scope="Application">
       <Value Profile="(Default)">20</Value>
+>>>>>>> refs/remotes/origin/master
```
Merge mistake

Commit:
Changes according to comments

Couldn't use MathHelper.Clamp, since for some reason it would break the
adjusting of the sound when at the min/max value

jensj12 commented on MeesGame/Gameplay/GameStates/LoadMenuState.cs:
```diff
@@ -64,10 +65,18 @@ public LoadMenuState()
+            {
+                GameEnvironment.GameStateManager.SwitchTo("EditorState");
+                //reset the gamestate to open blank;
+                GameEnvironment.GameStateManager.GetGameState("EditorState").Reset();
```
Use CurrentGameState, there's a property for that.

Commit:
Using CurrentGamestate property

Commit:
Removed unnecessary line of code

jensj12 commented:
Main menu and quit game buttons in settings menu don't work.

twjvdhorst commented:
Weird, they worked just fine before... I'll see what I can do

Commit:
Fixed buttons

Added the buttons directly to the overlay, as to fix the buttons

jensj12 committed:
Formatting

jensj12 commented:
MathHelper.Clamp works just fine.
I'd like to see the edit level button renamed to 'Level editor' and moved to the bottom of the screen. I've merged it for now so others can continue upon this as well. (create a new branch if you want to continue working on this)
